truffle/test/controller_deployment_interaction.js

```diff
@@ -0,0 +1,330 @@
+/* -------------------------------------------------------------------------- */
+
+const TuiChainController = artifacts.require("TuiChainController");
+const TuiChainMarket = artifacts.require("TuiChainMarket");
+const TuiChainLoan = artifacts.require("TuiChainLoan");
+
+const DaiMock = artifacts.require("DaiMock");
+
+/* -------------------------------------------------------------------------- */
+
+// Sequence of tests for the deployment and interaction with Controller Contract
+// This tests represent possible interactions from the backend with controller contract
+contract("Controller Deployment and Interaction", function (accounts) {
+
+
+  // market fee, is equivalent to 10% of a DAI
+  const marketFeeAttoDaiPerNanoDai = BigInt(Math.pow(10,8));
```

Comment:
Probably should be `BigInt(10) ** BigInt(8)`, ditto for all other arithmetic expressions.

Reply:
In this tests, the emitted events confirm that the values which are being passed are correct.
But I can change to your proposal if you feel more comfortable whit it.

Reply:
Would be clearer and prevent possible future confusion if we avoid floating-point arithmetic, even if it happens to work for the existing cases.

Reply:
👍